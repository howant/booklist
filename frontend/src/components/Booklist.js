import React from "react";

const Booklist = () => {
  return (
      <div className="App-booklist">
        <h2>To read</h2>
        <ul className="Books-to-read">
          <li>
            Lead with Respect
            <ul>
              <li>Michael Ballé & Freddy Ballé</li>
            </ul>
          </li>
          <li>
            Book 2
            <ul>
              <li>Author 1</li>
            </ul>
          </li>
          <li>
            Book 3
            <ul>
              <li>Author 3</li>
            </ul>
          </li>
        </ul>
        <h2>Currently reading</h2>
        <ul className="Books-reading">
          <li>
            Book 1
            <ul>
              <li>Author 1</li>
            </ul>
          </li>
        </ul>
        <h2>Finished reading</h2>
        <ul className="Books-finished">
          <li>
            Book 1
            <ul>
              <li>Author 1</li>
            </ul>
          </li>
          <li>
            Book 2
            <ul>
              <li>Author 2</li>
            </ul>
          </li>
        </ul>
      </div>
  );
};

export default Booklist;
