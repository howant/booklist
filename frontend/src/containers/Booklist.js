import { connect } from "react-redux";
import Booklist from "../components/Booklist";
import { requestBooks, requestSelectedBook } from "../actions";
import { push } from "connected-react-router";

const mapStateToProps = state => ({
  lists: state,
  loading: state.loading
});

const mapDispatchToProps = dispatch => ({
  requestBooks: books => dispatch(requestBooks(books)),
  requestSelectedBook: bookTitle => {
    dispatch(requestSelectedBook(bookTitle));
    dispatch(push('/booklist/description'));
  }
});

const BooklistContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Booklist);

export default BooklistContainer;
