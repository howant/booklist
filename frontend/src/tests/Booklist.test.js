import React from 'react';
import Booklist from '../components/Booklist';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const lists = {
    "To-read": [
      {
        "Title": "Lead With Respect",
        "Author": "Michael Ballé & Freddy Ballé"
      },
      {
        "Title": "Star Shroud",
        "Author": "Ken Lozito"
      },
      {
        "Title": "The Great Gatsby",
        "Author": "F. Scott Fitzgerald"
      }
    ],
    "Reading": {
      "Title": "Black Man",
      "Author": "Richard Morgan"
    },
    "Finished-reading": [
      {
        "Title": "Dark Eden",
        "Author": "Chris Beckett"
      },
      {
        "Title": "A Wild Sheep Chase",
        "Author": "Haruki Murakami"
      }
    ]
  };
  const tree = renderer
    .create(<Booklist lists={lists} requestBooks={jest.fn()}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
