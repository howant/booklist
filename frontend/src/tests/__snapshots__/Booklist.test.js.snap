// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App-booklist"
>
  <h2>
    To read
  </h2>
  <ul
    className="Books-to-read"
  >
    <li
      onClick={[Function]}
    >
      Lead With Respect
      <ul>
        <li>
          Michael Ballé & Freddy Ballé
        </li>
      </ul>
    </li>
    <li
      onClick={[Function]}
    >
      Star Shroud
      <ul>
        <li>
          Ken Lozito
        </li>
      </ul>
    </li>
    <li
      onClick={[Function]}
    >
      The Great Gatsby
      <ul>
        <li>
          F. Scott Fitzgerald
        </li>
      </ul>
    </li>
  </ul>
  <h2>
    Currently reading
  </h2>
  <ul
    className="Books-reading"
  >
    <li
      onClick={[Function]}
    >
      Black Man
      <ul>
        <li>
          Richard Morgan
        </li>
      </ul>
    </li>
  </ul>
  <h2>
    Finished reading
  </h2>
  <ul
    className="Books-finished"
  >
    <li
      onClick={[Function]}
    >
      Dark Eden
      <ul>
        <li>
          Chris Beckett
        </li>
      </ul>
    </li>
    <li
      onClick={[Function]}
    >
      A Wild Sheep Chase
      <ul>
        <li>
          Haruki Murakami
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
